src/render/filters.rs

```diff
@@ -274,6 +275,51 @@ impl ResolveFilter for DefaultIfNoneFilter {
     }
 }
 
+impl ResolveFilter for DateFilter {
+    fn resolve<'t, 'py>(
+        &self,
+        variable: Option<Content<'t, 'py>>,
+        py: Python<'py>,
+        template: TemplateString<'t>,
+        context: &mut Context,
+    ) -> ResolveResult<'t, 'py> {
+        let Some(value) = variable else {
+            return Ok(Some("".as_content()));
+        };
+
+        let fmt = match &self.argument {
+            Some(arg) => arg
+                .resolve(py, template, context, ResolveFailures::Raise)?
+                .expect("missing date format argument")
```

Comment:
We should explain why it's OK to use `expect`:

```suggestion
                .expect("missing argument in context should already have raised")
```